### PR TITLE
feat: Added support for RSA token signature and added JWKS endpoint

### PR DESCRIPTION
**Pull request by JigarJoshi**
- This PR adds the very basic functional RSA based RS256 algorithm to sign the JWT. 
- This also adds the JWKS endpoint.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -31,10 +37,59 @@ var bearerRegexp = regexp.MustCompile(`^(?:B|b)earer (\S+$)`)
...
+func (t *TokenSigner) init() {
+	if t.jwtConfig.RSAPrivateKey == "" && t.jwtConfig.Algorithm == jwa.RS256.String() {
+		log.Fatal("No RSA private key configured")
+	}
+	privateKeyData, err := os.ReadFile(t.jwtConfig.RSAPrivateKey)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	block, _ := pem.Decode(privateKeyData)
```
Comment (himank) [inline api/api.go:69]: check the error as well?

File: api/api.go [hunk 2]
```diff
@@ -31,10 +37,59 @@ var bearerRegexp = regexp.MustCompile(`^(?:B|b)earer (\S+$)`)
...
+	t := &TokenSigner{
+		jwtConfig: &config.JWT,
+	}
+	t.init()
+	return t
+}
+
+func (t *TokenSigner) init() {
+	if t.jwtConfig.RSAPrivateKey == "" && t.jwtConfig.Algorithm == jwa.RS256.String() {
+		log.Fatal("No RSA private key configured")
```
Comment (himank) [inline api/api.go:62]: instead of fatal in this method, you can do it at the caller and return an error from here.

File: api/jwks.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+	block, _ := pem.Decode(publicKeyPEM)
+	if block == nil || block.Type != "PUBLIC KEY" {
+		fmt.Println("failed to decode PEM block containing public key")
+		return nil, errors.New("Couldn't decode pem file")
+	}
+
+	publicKey, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		fmt.Println("failed to parse public key:", err)
```
Comment (himank) [inline api/jwks.go:53]: remove it

Comment (JigarJoshi) [reply api/jwks.go]: removed.

File: api/api.go (hunk 2 above)
Comment (JigarJoshi) [reply api/api.go:62]: yes good point.

File: api/api.go (hunk 1 above)
Comment (JigarJoshi) [reply api/api.go:69]: the `_` ignored part is not error it is remainder of bytes after reading block.

Comment (himank) [reply api/api.go:69]: ack

**Review by himank — APPROVED** [review]

**Pull request merged**